src/message.ts

```diff
@@ -142,3 +143,269 @@ export class MsgDelegate extends Msg {
     return true
   }
 }
+
+export class MsgCreateDataSource extends Msg {
+  owner: Address
+  name: string
+  description: string
+  script: string
+  sender: Address
+
+  constructor(
+    owner: Address,
+    name: string,
+    description: string,
+    script: string,
+    sender: Address,
+  ) {
+    super()
+    this.owner = owner
+    this.name = name
+    this.description = description
+    this.script = script
+    this.sender = sender
+  }
+
+  _replaceEmpty(placeholder: string, s?: string) {
+    if (s === undefined) return placeholder
+    return s
+  }
+
+  _readScript() {
+    return Buffer.from(fs.readFileSync(this.script, 'utf8')).toString('base64')
+  }
+
+  asJson() {
+    return {
+      type: 'oracle/CreateDataSource',
+      value: {
+        owner: this.owner.toAccBech32(),
+        name: this.name,
+        description: this._replaceEmpty('TBA', this.description),
+        executable: this._readScript(),
+        sender: this.sender.toAccBech32(),
+      },
+    }
+  }
+
+  getSender() {
+    return this.sender
+  }
+
+  validate() {
+    if (this.name.length <= 0) throw Error('Invalid data source name')
```

Comment:
Let's create custom error for this like valuator 
[Ref](https://github.com/bandprotocol/bandchain.js/pull/7)

Reply:
done